### ISSUE TITLE
Issue 02: coarse train config で loss enabled なのに対応 trace head disabled の設定を早期 reject する

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#383]
# Issue 02: coarse train config で loss enabled なのに対応 trace head disabled の設定を早期 reject する

## 背景

現在の coarse training config では、以下のような不整合 config が通り得る。

```yaml
train:
  pickability_pu:
    enabled: true
model:
  trace_heads:
    pickability:
      enabled: false
```

または、

```yaml
train:
  coarse_usability:
    enabled: true
model:
  trace_heads:
    coarse_usable:
      enabled: false
```

この場合、loss 計算時に必要な logits が model output に存在せず、最初の batch で `KeyError` などになって落ちる。Codex / CI / 実運用では、config load 時点で分かりやすく落としたい。

## 目的

`load_coarse_train_config()` で、train loss と trace head の整合性を検証する。

## 対象ファイル

主対象:

```text
packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/config.py
```

テスト対象:

```text
packages/seisai-engine/tests/test_fbpick_coarse_config_trace_heads.py
```

## 実装方針

`load_coarse_train_config()` で以下を検証する。

```python
if pickability_pu.enabled and not trace_heads.pickability_enabled:
    raise ValueError(
        "train.pickability_pu.enabled requires "
        "model.trace_heads.pickability.enabled"
    )

if coarse_usability.enabled and not trace_heads.coarse_usable_enabled:
    raise ValueError(
        "train.coarse_usability.enabled requires "
        "model.trace_heads.coarse_usable.enabled"
    )
```

検証タイミングは、`trace_heads`、`pickability_pu`、`coarse_usability` を読み込んだ直後がよい。

現在すでにある以下の検証は残す。

```text
train.coarse_usability.enabled is true
  -> paths.coarse_usability_label_files is required
```

### reject する設定

```yaml
train:
  pickability_pu:
    enabled: true
model:
  trace_heads:
    pickability:
      enabled: false
```

```yaml
train:
  coarse_usability:
    enabled: true
model:
  trace_heads:
    coarse_usable:
      enabled: false
paths:
  coarse_usability_label_files:
    - labels.npz
```

### allow する設定

head は enabled だが loss が disabled の設定は許容する。

```yaml
model:
  trace_heads:
    pickability:
      enabled: true
train:
  pickability_pu:
    enabled: false
```

理由: 学習済み head 付き checkpoint の inference 用 config、または段階的 fine-tuning のために head だけ有効にする運用を壊さないため。

## 受け入れ条件

- `train.pickability_pu.enabled: true` かつ `model.trace_heads.pickability.enabled: false` が `load_coarse_train_config()` で `ValueError` になる。
- `train.coarse_usability.enabled: true` かつ `model.trace_heads.coarse_usable.enabled: false` が `load_coarse_train_config()` で `ValueError` になる。
- error message に、どの train loss とどの trace head が不整合かが明記される。
- 既存の valid config は通る。
- loss 実行時の `KeyError` に到達する前に config load で落ちる。

## 追加テスト

`packages/seisai-engine/tests/test_fbpick_coarse_config_trace_heads.py` に以下を追加する。

### test: pickability PU loss requires pickability head

- `_cfg(tmp_path)` を使う。
- `cfg['train']['pickability_pu'] = {'enabled': True, 'class_prior': 0.6}` を設定。
- `model.trace_heads.pickability.enabled` は設定しない、または false にする。
- `load_coarse_train_config(cfg)` が `ValueError` を出すこと。
- message に `train.pickability_pu.enabled` と `model.trace_heads.pickability.enabled` が含まれること。

### test: coarse usability loss requires coarse usable head

- `_cfg(tmp_path)` を使う。
- `cfg['paths']['coarse_usability_label_files'] = ['labels.npz']` を設定。
- `cfg['train']['coarse_usability'] = {'enabled': True}` を設定。
- `model.trace_heads.coarse_usable.enabled` は設定しない、または false にする。
- `load_coarse_train_config(cfg)` が `ValueError` を出すこと。
- message に `train.coarse_usability.enabled` と `model.trace_heads.coarse_usable.enabled` が含まれること。

### test: enabled head with disabled loss remains valid

- pickability head enabled、PU loss disabled の config が通ること。
- coarse_usable head enabled、usability loss disabled の config も label file 不要で通ることを確認する。

## 対象外

- inference config の検証。これは別 issue で扱う。
- site54 config generator の CLI guard。これは別 issue で扱う。
- PU loss の数式変更。
